src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,43 @@
 package core.basesyntax;
 
-class Pair {
+import java.util.Objects;
 
+class Pair<T, V> {
+    private T firstElement;
+    private V secondElement;
+
+    private Pair(T i, V s) {
+        this.firstElement = i;
+        this.secondElement = s;
+    }
+
+    public T getFirst() {
+        return firstElement;
+    }
+
+    public V getSecond() {
+        return secondElement;
+    }
+
+    public static <T, V> Pair of(T firstElement, V secondElement) {
+        return new Pair<>(firstElement, secondElement);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Pair)) {
```

Comment:
Remember about all rules for overriding equals method and use `getClass()` instead of `instanceof`